### DISCUSSION TITLE
contactBreakingThreshold random behaviour

### DISCUSSION THREAD

**Discussion by mostafa-hussein** [bulletphysics/bullet3#3364] [General]
Hi, We are trying to force Bullet/PyBullet to consider adding contact constraints (we already changed the solver to create LCP) between objects with large distance between them. I changed this parameter (contactBreakingThreshold) and also this (ContactProcessingthreshold) but when we do this some of the objects start to move randomly. Especially when we change the contactBreakingThreshold option.
Does any one have any idea how to fix this or even debug it ?

**Comment by erwincoumans** [comment]
Don't have the time to dig into at the moment. Have you tried leaving the contactBreakingThreshold default, and only modify the ContactProcessingThreshold?

**Reply by mostafa-hussein** [reply]
Hi,
Thanks for your time I can't imagine how busy you are.
I think I tried most of the combination of doing this. The problem is it works good with simple test like having a sphere and a plane, but when I included a robot (Kuka) and a cube for example in a pushing environment it won't work.
When I change the ContactProcessingThreshold only the random behavior disappears and work normal but the number of constraints keep changing based on the distance.

**Comment by erwincoumans** [comment]
I am working on a new physics engine, Tiny Differentiable SImulator, there you can put all contacts in the LCP solver.
https://github.com/google-research/tiny-differentiable-simulator/blob/master/src/mb_constraint_solver.hpp

**Reply by mostafa-hussein** [reply]
That's sound interesting I sure will take a look on it. Thanks.
One last question, Do you think that we can fix the number of constraints in the simulator each time step by increasing the distance or other parameter or not ?
